mathics/builtin/assignments/internals.py

```diff
@@ -122,6 +124,32 @@ def is_protected(tag, defin):
     return A_PROTECTED & defin.get_attributes(tag)
 
 
+def normalize_lhs(lhs, evaluation):
```

Comment:
If seems to me that this function and, more generally, the entire module should be moved to `mathics.core`.

Reply:
Suggested function signature with type annotation: 

```
def normalize_lhs(lhs: Expression, evaluation: Evaluation) -> Tuple(Expression, str):
```

Reply:
By the way, although there are lots of little comments, this style of function is a _great_ improvement over the style that we were left with. Thanks!